queue/QueueConsumer.js

```diff
@@ -1,16 +1,48 @@
 const BaseQueue = require('./BaseQueue')
 const generateTagName = require('./util/generateTagName')
+const assertQueue = require('./util/assertQueue')
+
+const DEFAULT_PREFETCH_COUNT = 10
+const DEFAULT_MAX_MESSAGE_RETRIES = 5
+const DEAD_LETTER_HEADER = 'x-death'
+
+const FANOUT = 'fanout'
+const REQUEUE_MESSAGE = true
+const REJECT_ALL_MESSAGES = true
+
+const REJECTED_REASON = 'rejected'
 
 module.exports = class QueueConsumer extends BaseQueue {
-  constructor ({queueName, connection, prefetchCount, logger}) {
+  constructor (options) {
+    const {
+      queueName,
+      connection,
+      prefetchCount,
+      logger,
+      maxMessageRejections,
+      deadLetterQueueName
+    } = options
+
     super({
       queueName,
       connection,
       tag: generateTagName(queueName, 'consumer'),
       logger
     })
 
-    this._prefetchCount = prefetchCount || 10
+    this._prefetchCount = prefetchCount || DEFAULT_PREFETCH_COUNT
+    this._deadLetterExchangeName = `${queueName}-dead-letter-exchange`
+
+    this._deadLetterQueueName = deadLetterQueueName ||
+      `${queueName}-dead-letter-queue`
+
+    this._maxMessageRejections = isNaN(maxMessageRejections)
+      ? DEFAULT_MAX_MESSAGE_RETRIES
+      : +maxMessageRejections
```

Comment:
`isNaN(null) === false` and `+null === 0`. This will set `this._maxMessageRejections = 0` if null is passed in. Maybe use `this._maxMessageRejections = Number.isInteger(maxMessageRejections) ? maxMessageRejections : DEFAULT_MAX_MESSAGE_RETRIES`? Also has the added benefit of avoided the unary.